component/component/main.jsonnet

```diff
@@ -28,6 +28,78 @@ local secrets = [
   for s in std.objectFields(params.secrets)
 ];
 
+local cronjob(name, subcommand, schedule) = {
+  kind: 'CronJob',
+  apiVersion: 'batch/v1',
+  metadata: {
+    name: name,
+    namespace: paramsACR.namespace,
+    labels+: labels,
+  },
+  spec: {
+    concurrencyPolicy: 'Forbid',
+    failedJobsHistoryLimit: 5,
+    jobTemplate: {
+      spec: {
+        template: {
+          spec: {
+            restartPolicy: 'OnFailure',
+            containers: [
+              {
+                name: 'exoscale-metrics-collector-backfill',
+                image: collectorImage,
+                args: [
+                  subcommand,
+                ],
+                envFrom: [
+                  {
+                    secretRef: {
+                      name: credentials_secret_name,
+                    },
+                  },
+                ],
+                env: [
+                  {
+                    name: 'password',
+                    valueFrom: {
+                      secretKeyRef: {
+                        key: 'password',
+                        name: 'reporting-db',
+                      },
+                    },
+                  },
+                  {
+                    name: 'username',
+                    valueFrom: {
+                      secretKeyRef: {
+                        key: 'username',
+                        name: 'reporting-db',
+                      },
+                    },
+                  },
+                  {
+                    name: 'ACR_DB_URL',
+                    value: 'postgres://$(username):$(password)@%(host)s:%(port)s/%(name)s?%(parameters)s' % paramsACR.database,
+                  },
+                ],
+                resources: {},
+              },
+            ],
+          },
+        },
+      },
+    },
+    schedule: schedule,
+    successfulJobsHistoryLimit: 3,
+  },
+};
+
+local cronjobs = [
+    cronjob(alias + '-objectstorage', 'objectstorage', params.schedule),
+    if params.dbaasEnabled then
+        cronjob(alias + '-dbaas', 'dbaas', params.dbaasSchedule),
+];
```

Comment:
that's ugly - is there a better way?

Reply:
I think it's fine as is. You could probably make it a bit cleaner, but you'll probably lose clarity so I think this is fine

Reply:
You could get rid of the null check below by doing something like:

```
local cronjobs = [cronjob(..)] + 
if enabled then 
  [cronjob(..)]
else []
```
Not sure how much better that is though.

Reply:
thanks. Yeah I think that's not really better. Will leave as-is